src/filters/oidc/token_response.cc

```diff
@@ -35,17 +47,15 @@ absl::optional<TokenResponse> TokenResponseParserImpl::Parse(
   auto fields = message.fields();
   // https://openid.net/specs/openid-connect-core-1_0.html#TokenResponse
   // token_type must be Bearer
-  auto token_type = fields.find("token_type");
+  auto token_type = fields.find(token_type_field);
   if (token_type == fields.end() ||
-      !(token_type->second.string_value() == "Bearer" ||
-        token_type->second.string_value() ==
-            "bearer")) {  // We allow either Bearer or bearer
+      !(token_type->second.string_value() == bearer_token_type)) {
     spdlog::info("{}: missing or incorrect `token_type` in token response",
                  __func__);
     return absl::nullopt;
```

Comment:
@nickrmc83 Our test OIDC provider fails to pass this validation because it returns the lowercase `bearer` as `token_type`. We propose that we treat this `token_type` value in a case-insensitive manner (aka, both `Bearer` and `bearer` are okay).

We researched both OIDC and Oauth 2 specs, which do not conclusively dictate whether the `token_type` value is case sensitive or insensitive. Here are a few sources:

1. [Oauth 2 spec](https://tools.ietf.org/html/rfc6749#section-4.2.2):
```
   token_type
         REQUIRED.  The type of the token issued as described in
         Section 7.1.  Value is case insensitive.
```
Which says the value is case insensitive, which is also supported by the example given in [this source](https://www.oauth.com/oauth2-servers/access-tokens/access-token-response/).

2. Meanwhile, the [OIDC spec](https://openid.net/specs/openid-connect-core-1_0.html#TokenResponse) says `The OAuth 2.0 token_type response parameter value MUST be Bearer, as specified in OAuth 2.0 Bearer Token Usage [RFC6750]`, which does indicates that it must be `Bearer`. But then again, the spec here is citing OAuth 2 spec, which, as mentioned above, says the value can be insensitive.

Reply:
I've reinstated the case insensitive comparison in f7bc68875f3384b172dcecd24072987e42f22fb3